File: ButterKnifeExample/app/src/main/java/com/example/tengfei/MainActivity.java
package com.example.tengfei;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.TextView;

import com.example.annotations.BindView;
import com.example.butterknife.ButterKnifeUtils;

/**
 * @author tengfei
 */
public class MainActivity extends AppCompatActivity {

    @BindView(R.id.main_text)
    TextView textView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ButterKnifeUtils.init(this);

        textView.setText("hahahha");

    }
}

File: CoroutineSample/app/src/main/java/com/example/coroutine/MainActivity.kt
package com.example.coroutine

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.launch
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

class MainActivity : AppCompatActivity() {

    val tag = "MainActivity.test"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        bt.setOnClickListener {
            GlobalScope.launch {
                val responseBody = getUser().getUser("bennyhuo")
                Log.i(tag, responseBody.string())
            }
        }


    }

    private fun getUser(): GithubApi {
        val retrofit = Retrofit.Builder()
                .addConverterFactory(GsonConverterFactory.create())
                .baseUrl("https://api.github.com")
                .build()
        return retrofit.create(GithubApi::class.java)
    }
}
